cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -179,7 +178,8 @@ CUDF_TYPE_MAPPING(cudf::duration_s, type_id::DURATION_SECONDS)
 CUDF_TYPE_MAPPING(cudf::duration_ms, type_id::DURATION_MILLISECONDS)
 CUDF_TYPE_MAPPING(cudf::duration_us, type_id::DURATION_MICROSECONDS)
 CUDF_TYPE_MAPPING(cudf::duration_ns, type_id::DURATION_NANOSECONDS)
-CUDF_TYPE_MAPPING(dictionary32, type_id::DICTIONARY32)
+CUDF_TYPE_MAPPING(cudf::dictionary32, type_id::DICTIONARY32)
+CUDF_TYPE_MAPPING(cudf::string_view, type_id::STRING)
 CUDF_TYPE_MAPPING(cudf::list_view, type_id::LIST)
 CUDF_TYPE_MAPPING(numeric::decimal32, type_id::DECIMAL32)
```

Comment:
Why are the decimal types defined in a top-level namespace `numeric` and not in the namespace `cudf`? It's not even `cudf::numeric`, just `::numeric`. 😕  I think this shows up in the stringified type name, so I feel uncomfortable about the namespace not being in `cudf`.

Reply:
cc: @codereport @jrhemstad if you can shed any light on this.

Reply:
If cudf used `cuda::std::string_view` as-is instead of it's own custom `string_view` type, would you feel we'd need to alias that into the `cudf` namespace? 

It was intentional to treat the `decimalXX` types as if they were provided by an external library.

Reply:
I think if that’s the intent, we should create a directory `cpp/include/numeric` and move these files into it.

Reply:
I am strongly against `cpp/include/numeric` as that would install as `include/numeric/` and that will conflict with other projects named `numeric` or a file named `numeric` in the include dir.

We need a better name or prefix it with `nv_` or `rapids_`

Reply:
I think `numeric::` as a namespace is too broad, regardless of what files it lives in. I would strongly prefer to have everything in `cpp/include/cudf` living in the `cudf::` namespace. I think that's just a rule of good packaging. If someone wants to make a separate library at some future point, the files can be removed and taken out of that namespace.

Reply:
> as if they were provided by an external library

This "as if" just isn't true in the current state of things -- the implication that these are (or could be) externally provided is more detrimental to developer/user understanding than beneficial to "separation of concerns", in my eyes. A self-contained sub-namespace like `cudf::fixed_point` or `cudf::numeric` would be a sufficient solution that maintains a degree of separation while clearly indicating that the types are defined and owned by `cudf::`.

Reply:
I'll defer this discussion to an issue so we don't block this PR on an unrelated question. https://github.com/rapidsai/cudf/issues/12172